test/dymaptic.GeoBlazor.Core.Test.Blazor.Shared/Components/FeatureLayerViewTests.razor

```diff
@@ -58,4 +58,194 @@
+                new Point(1, 1),
+                attributes: new AttributesDictionary(new Dictionary<string, object?> { ["OBJECTID"] = 2 })
+            )
+        };
+
+        AddMapRenderFragment(
+            @<MapView class="map-view" OnViewRendered="renderHandler" OnLayerViewCreate="OnLayerViewCreate">
+                <Map>
+                    <Basemap><BasemapStyle Name="BasemapStyleName.ArcgisTopographicBase" /></Basemap>
+                    <FeatureLayer @ref="layer"
+                                  Source="graphics"
+                                  OutFields="@(["*"])"
+                                  SpatialReference="SpatialReference.Wgs84"
+                                  GeometryType="FeatureGeometryType.Point"
+                                  ObjectIdField="OBJECTID" />
+                </Map>
+            </MapView>);
+
+        await WaitForMapToRender();
+
+        // Wait for the FeatureLayerView
+        for (var i = 0; layerView is null && i < 100; i++) { await Task.Delay(50); }
+        Assert.IsNotNull(layerView, "FeatureLayerView did not materialize");
+
+        // ---------------------------
+        // Show that getters reflect JS state
+        // ---------------------------
+
+        // 1) A settable prop: change it and confirm JS & getter agree (proves getter isn't stale)
+        await layerView!.SetVisible(false);
+        await AssertJavaScript("assertObjectHasPropertyWithValue", args: [layerView.Id, "visible", false]);
+        var visible = await layerView.GetVisible();
+        Assert.AreEqual(false, visible);
+
+        await layerView.SetVisible(true);
+        await AssertJavaScript("assertObjectHasPropertyWithValue", args: [layerView.Id, "visible", true]);
+        visible = await layerView.GetVisible();
+        Assert.AreEqual(true, visible);
+
+        // 2) Read-only props: read them, then assert JS reports the same value
+        var max = await layerView.GetMaximumNumberOfFeatures();
+        Assert.IsTrue(max >= 0);
+        await AssertJavaScript("assertObjectHasPropertyWithValue", args: [layerView.Id, "maximumNumberOfFeatures", max]);
+
+        var exceeded = await layerView.GetMaximumNumberOfFeaturesExceeded();
+        await AssertJavaScript("assertObjectHasPropertyWithValue", args: [layerView.Id, "maximumNumberOfFeaturesExceeded", exceeded]);
+
+        // 3) Complex object roundtrip: ensure we can set/get an object without touching component-only internals
+        var hl = new HighlightOptions(color: new MapColor("cyan"), fillOpacity: 1.0, haloOpacity: 1.0);
+        await layerView.SetHighlightOptions(hl);
+        var gotHl = await layerView.GetHighlightOptions();
+        Assert.IsNotNull(gotHl);
+
+        // Optional: feature effect as an object (do not set IncludedEffect here)
+        var fx = new FeatureEffect();
+        await layerView.SetFeatureEffect(fx);
+        var gotFx = await layerView.GetFeatureEffect();
+        Assert.IsNotNull(gotFx);
```

Comment:
The only part of this test that isn't a duplicate of the previous test is the last four lines. Can you reorganize to remove the duplication?